### PR TITLE
DROID-3006 Create object | Fix | Crash on type selection

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt [hunk 1]
```diff
@@ -1775,7 +1775,6 @@ class HomeScreenViewModel(
 
     fun onNewWidgetSourceTypeSelected(
         type: ObjectWrapper.Type,
-        space: SpaceId,
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:1778]: it is not used?

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt [hunk 2]
```diff
@@ -1807,7 +1806,6 @@ class HomeScreenViewModel(
 
     fun onCreateObjectForWidget(
         type: ObjectWrapper.Type,
-        widget: Id,
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:1810]: idem?

**Review by uburoiubu — APPROVED** [review]

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt (hunk 1 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:1778]: no, this method uses spaceManager.get

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt (hunk 2 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:1810]: same

File: app/src/main/java/com/anytypeio/anytype/ui/objects/types/pickers/ObjectSelectTypeFragment.kt [hunk 3]
```diff
@@ -26,8 +26,8 @@ class ObjectSelectTypeFragment : BaseObjectTypeChangeFragment() {
     }
 
     override fun onItemClicked(item: ObjectWrapper.Type) {
-        withParent<OnObjectSelectTypeAction> {
-            onProceedWithUpdateType(item)
+        withParent<ObjectTypeSelectionListener> {
```
[the remark this answers is not in the document]
Comment (konstantiniiv) [reply app/src/main/java/com/anytypeio/anytype/ui/objects/types/pickers/ObjectSelectTypeFragment.kt]: done

**Pull request merged**